src/plugins/fs/reiserfs.c

```diff
@@ -0,0 +1,386 @@
+ * @device: the device containing the file system to set label for
+ * @label: label to set
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the label of reiserfs file system on the @device was
+ *          successfully set or not
+ *
+ * Tech category: %BD_FS_TECH_REISERFS-%BD_FS_TECH_MODE_SET_LABEL
+ */
+gboolean bd_fs_reiserfs_set_label (const gchar *device, const gchar *label, GError **error) {
+    const gchar *args[5] = {"reiserfstune", "-l", label, device, NULL};
+
+    if (!check_deps (&avail_deps, DEPS_REISERFSTUNE_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    if (strlen (label) > 16)
+        bd_utils_log (BD_UTILS_LOG_WARNING, "Given label is too long for ReiserFS and will be truncated." \
+                                            "Labels on ReiserFS can be at most 16 characters long");
+
+    return bd_utils_exec_and_report_error (args, NULL, error);
+}
+
+/**
+ * bd_fs_reiserfs_set_uuid:
+ * @device: the device containing the file system to set UUID for
+ * @uuid: (allow-none): UUID to set or %NULL to generate a new one
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether the uuid of reiserfs file system on the @device was
+ *          successfully set or not
+ *
+ * Tech category: %BD_FS_TECH_REISERFS-%BD_FS_TECH_MODE_SET_UUID
+ */
+gboolean bd_fs_reiserfs_set_uuid (const gchar *device, const gchar *uuid, GError **error) {
+    const gchar *args[5] = {"reiserfstune", "-u", uuid, device, NULL};
+
+    if (!check_deps (&avail_deps, DEPS_REISERFSTUNE_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    if (!uuid)
+        args[2] = "random";
+
+    return bd_utils_exec_and_report_error (args, NULL, error);
+}
+
+/**
+ * bd_fs_reiserfs_get_info:
+ * @device: the device containing the file system to get info for
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: (transfer full): information about the file system on @device or
+ *                           %NULL in case of error
+ *
+ * Tech category: %BD_FS_TECH_REISERFS-%BD_FS_TECH_MODE_QUERY
+ */
+BDFSReiserFSInfo* bd_fs_reiserfs_get_info (const gchar *device, GError **error) {
+    const gchar *args[3] = {"debugreiserfs", device, NULL};
+    gboolean success = FALSE;
+    gchar *output = NULL;
+    BDFSReiserFSInfo *ret = NULL;
+    gchar **lines = NULL;
+    gchar **line_p = NULL;
+    gchar *val_start = NULL;
+
+    if (!check_deps (&avail_deps, DEPS_DEBUGREISERFS_MASK, deps, DEPS_LAST, &deps_check_lock, error))
+        return FALSE;
+
+    success = bd_utils_exec_and_capture_output (args, NULL, &output, error);
+    if (!success)
+        /* error is already populated */
+        return FALSE;
+
+    ret = g_new0 (BDFSReiserFSInfo, 1);
+
+    success = get_uuid_label (device, &(ret->uuid), &(ret->label), error);
+    if (!success) {
+        /* error is already populated */
+        bd_fs_reiserfs_info_free (ret);
+        return NULL;
+    }
```

Comment:
This block getting the label should be done before we fork&exec the binary, it's cheaper. And the code would be better readable, I think.

Reply:
Ok. I thought starting with something more likely to fail is better, but I do it your way.